public/js/admin.js

```diff
@@ -47,6 +47,11 @@
     $("#messageDisplay").show().one();
   });
 
+  socket.on('expired token', function(service){
+    alert('Your session has expired. Please log in again.');
+    window.location.replace('https://cas.byu.edu/cas/logout');
+    window.location.replace('https://cas.byu.edu/cas/login?service=' + service);
```

Comment:
I haven't pulled this down to check but does this wait for the logout to finish then redirect to login?